tests/cpu_bugs/xen_hvm.pm

```diff
@@ -0,0 +1,370 @@
+#			}
+#		}
+#	}
+# }
+#
+#  _________________________________________________      ___________________________________________________
+# |Start/check the parameters in the mitigation_list| => |Input the parameter=value into the cmdline settings|
+#  -------------------------------------------------     |Then grub_config and reboot.                       |
+#                                                         ---------------------------------------------------
+#                                                                               /\
+#                                                                               ||
+#                                                                               ||
+#                                                                               ||
+#                                                                               ||
+#  _________________________________________              ______________________\/_________________________
+# |Recall the infos after success or failure|   <======  |check the outcomes based on the value's key value|
+#  -----------------------------------------              -------------------------------------------------
+#                 /\
+#                 ||
+#                 ||
+#                 ||
+#                 ||
+#  _______________\/_________________
+# |Remove the cmdline and grub_config.|
+# |And pick the next para=value.      |_____________________________________________________
+# |When pick the last key value in a dic, jump to the next para and repeat the process above.|
+#  -----------------------------------------------------------------------------------------
+#
+#
+# Put into pti, l1tf, mds into this structure. and The testing code will read them all and
+# one by one execution checking.
+#
+
+package xen_hvm;
+use strict;
+use warnings;
+use base "consoletest";
+use bootloader_setup;
+use Mitigation;
+use ipmi_backend_utils;
+use power_action_utils 'power_action';
+use testapi;
+use utils;
+
+my $qa_password                     = get_required_var("QA_PASSWORD");
+my $git_repo_url                    = get_required_var("MITIGATION_GIT_REPO");
+my $hvm_domain_name                 = get_var('HVM_DOMAIN_NAME');
+my $install_media_url               = get_required_var('INSTALL_MEDIA');
+my $hvm_vm_install_kernel_parameter = get_required_var('HVM_VM_INSTALL_KERNEL_PARAMETER');
+my $git_branch_name                 = get_required_var("MITIGATION_GIT_BRANCH_NAME");
+my $hvm_test_mode                   = get_required_var('HVM_TEST_MODE');
+my $hvm_single_test                 = get_required_var('HVM_SINGLE_TEST');
+my $vm_ip_addr;
+my $add_cmdline_settings = "mds=full";
+my $vm_domain_name;
+my $complete_cmd;
+my $hvm_imagepool_path = "/xen_hvm";
+
+my $hypervisor_mitigations_list = {
+    'spec-ctrl' => {
+        #This is hypervisor fully disable situation.
+        no => {
+            default => {
+                nextmove => {'xpti=false'}
```

Comment:
nextmove is the extra parameters for spec-ctrl ? if yes, xpti=false is meaningless when "spect-ctrl" is assigned to "no"

Reply:
When we discussed before, you said that we need to set spect-ctrl and xpti at the same time when in disable mode. So now just have to delete the xpti=false, right?

Reply:
I think you misunderstand my meaning before,  i mean pti can be set alone on hvm whaterver xen hypervisor setting. For PV inside, could  ignore pti setting, because the xen hypervisor mititgation ability can extend to guest.

I think it should be removed.

Reply:
OK